### ISSUE TITLE
RSession crash due to unexpected exception: vector::_M_range_check

### ISSUE THREAD

**Issue by zachhannum** [rstudio/rstudio#15365]
### System details

    RStudio Edition : 2024.04
    RStudio Version : 
    OS Version      : 
    R Version       : 

### Steps to reproduce the problem

This issue comes from support ticket # 110890.

1. Start an RStudio IDE session.

2. Create an R Script with the following code:

  ```
rm(list = ls())

control = data.frame(lopnr=1:1e6,is_female=sample(c(0,1),1e6,replace=T))
print(nrow(control))
sink(file='loop.out')
for (i in 1:nrow(control)){
dat<-control[i, ]
print(dat)
flush.console()
}
sink(file=NULL)
  ```

3. Close the browser and wait a few minutes
4. Attempt to reopen the session and observe that it has failed/crashed in some way

### Describe the problem in detail

I ran the same script on the same test machine using `RScript` and it was able to complete successfully. I wasn't able to reproduce the _exact_ failure as shown in the ticket; in my environment the entire session fails with some kind of seg fault, but I suspect it's still the same underlying issue.

### Describe the behavior you expected

The session should not crash.

- [ ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ ] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by jlsm-se** [comment]
Duplicate of #15330 ?

**Comment by jeffvroom** [comment]
Or I was thinking: https://github.com/rstudio/rstudio/issues/14643

**Comment by ronblum** [comment]
@ronblum Check if fixed per #15330.

**Comment by ronblum** [comment]
Verified fixed in RStudio Server 2024.12.0+456 on Red Hat 9 via Chrome on Windows 11.

```
> source("~/bobo.R")
[1] 1000000
Connected to your session in progress, last started 2024-Nov-25 16:53:17 UTC (12 minutes ago)
> x <- 1
>
```
